testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/settings/AppSettings.kt

```diff
@@ -170,7 +170,7 @@ private class Settings(context: Context) {
             prefs.edit().putString(CONTACT_NAMES, Gson().toJson(contactNames)).apply()
 
     val isContactsSaved: Boolean
-        get() = prefs.contains(CONTACT_NAMES)
+        get() = prefs.contains(CONTACT_NAMES) && !contactNames.isNullOrEmpty()
```

Comment:
I think we should not add the `&& !contactNames.isNullOrEmpty()` here. The contact list can be checked separately when the user gets the list from sharedpreference.